### PR TITLE
Object: Remove error on disconnect of locked signals

### LINKED ISSUE #34650: Editor unresponsive when removing SoftBody node
**Godot version:** v3.2.beta.custom_build.318c69351

**OS/device including version:** macOS Catalina 10.15.1 / MacBook Pro (Retina, 13-inch, Early 2015)

**Issue description:**

When I delete a `SoftBody` node, the editor becomes unresponsive. It looks like an infinite loop, printing tons of the following error:
```
ERROR: _disconnect: Attempt to disconnect signal 'node_removed' while in emission callback. Use CONNECT_DEFERRED (to be able to safely disconnect) or CONNECT_ONESHOT (for automatic disconnection) as connection flags.
   At: core/object.cpp:1522.
```

**Steps to reproduce:**
1. Create a new 3D scene
2. Add a `SoftBody` node as the child of the default `Spatial` node
3. Assign the `Spatial` node to the `SoftBody`'s `Parent Collision Ignore` property
4. Right click the `SoftBody` and select "Delete Node(s)"
5. Click "OK" when prompted

---

**Update:** I just tried the same thing with `InterpolatedCamera`. Assign the parent Spatial node to its `Target` property and try to delete the node, the editor became unresponsive too. (with the same error output.)

### LINKED ISSUE #34769: Editor freeze after setting focus neighbor and changing scene tab
**Godot version:**
27b74f42b

**Issue description:**
So uh... See the steps.

**Minimal reproduction project:**
[ReproductionProject.zip](https://github.com/godotengine/godot/files/4017866/ReproductionProject.zip)

**Steps to reproduce:**
1. Open the minimal project
2. Select Name node
3. Set `focus_next` to whatever node
4. Open the second scene
5. The editor wi-

### PR DESCRIPTION
**Pull request by akien-mga**
According to https://github.com/godotengine/godot/commit/22637beb2ed625c3e43ab75ab5865b57d7470948#commitcomment-36651823
and as confirmed by @reduz, this seems not to be necessary now that we
copy-on-write.

This triggered freeze scenarios in cases where a node would be deleted
while being used as a target in a signal emission.

Fixes #34650.
Fixes #34769.

Now those two errors go back to reporting:
```
ERROR: emit_signal: Condition ' !target ' is true. Continuing..:
   At: core/object.cpp:1191.
```

CC @timothyqiu

### CODE REVIEW

File: core/object.cpp [hunk 1]
```diff
@@ -1519,10 +1519,6 @@ void Object::_disconnect(const StringName &p_signal, Object *p_to_object, const
 	Signal *s = signal_map.getptr(p_signal);
 	ERR_FAIL_COND_MSG(!s, vformat("Nonexistent signal '%s' in %s.", p_signal, to_string()));
 
-	ERR_FAIL_COND_MSG(s->lock > 0,
-			vformat("Attempt to disconnect %s signal '%s' while in emission callback '%s' (in target %s). Use CONNECT_DEFERRED (to be able to safely disconnect) or CONNECT_ONESHOT (for automatic disconnection) as connection flags.",
```
Comment (akien-mga) [inline core/object.cpp]: I had made this error message so nice and explicit :'(

**Comment by akien-mga** [comment]
The second commit also solves the errors printed:
```
ERROR: emit_signal: Condition ' !target ' is true. Continuing..:
   At: core/object.cpp:1191.
```

I kept it separate as it's not necessary for the fix, and the errors pre-existed commit 22637be.

**Comment by Faless** [comment]
This seems to cause a regression for #33290 (bringing back the read after free) , as #33788 no longer applies.

**Comment by MightyPrinny** [comment]
I think this fixes #35014

**Pull request merged**